server/SocketManager/socketmanager.js

```diff
@@ -8,4 +10,37 @@ const removeSocket = usertoken => {
   delete sockets[usertoken];
 };
 
-export { sockets, updateSocket, removeSocket };
+const emitMessage = (socket, message, data) => {
+  if (socket) {
+    socket.emit(message, data);
+  }
+};
+
+export default function initializeSocket(server) {
+  const io = new Server(server, {
+    path: '/hv/socket',
+    cors: {
+      origin: '*',
+    },
+  });
+
+  // Listen for when the client connects via socket.io-client
+  io.on('connection', socket => {
+    console.warn(`User connected ${socket.id}`);
+
+    socket.on('addUser', user => {
+      let usertoken = user.socket_id;
+      updateSocket(usertoken, socket);
+      socket.emit('AddedUser', {
```

Comment:
In the `addUser` event handler, consider adding error handling for cases where the `usertoken` might be undefined or invalid. This will help avoid issues when trying to update the socket.
*This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*